src/main/java/org/owasp/validator/css/CssHandler.java

```diff
@@ -174,7 +174,7 @@ public String getCleanStylesheet() {
 	 * @return the error messages generated during parsing
 	 */
 	public Collection<String> getErrorMessages() {
-		return new ArrayList<String>(errorMessages);
+		return new ArrayList<>(errorMessages);
```

Comment:
Hi, @davewichers  I just want to reduce the code, as well as the warnings during the IDEA compilation period.
Reason:
Generic types only appear during static type checking, after which all generic types in the program are erased and replaced with their non-generic upper bounds, which means that when you're using generics, the only thing you know is that you're using an `Object`.
For example, `List<String>` and `List<Integer>` are actually of the same type at runtime, so at java7, the generics that follow can be omitted because they are no longer meaningful.
```diff
- List<String> list = new ArrayList<String>(); 
+ List<String> list = new ArrayList<>(); 
```

Reply:
OK - But we are likely to move to Java 8+ soon, so wouldn't it be better to leave them in?

Reply:
As far as I know, Oracle currently has no plan to discard generic erasure, so in the version after java7, idea still has this warning.
Of course, deletion and retention will not affect the use of the software.
@spassarop, what do you think of this?

Reply:
This warning is of type: redundant type arguments, in Eclipse. This warning is OFF by default in Eclipse which is why I've never seen it. Given that it OFF by default, and I personally like this style, I'd leave these alone unless @spassarop really thinks we should remove them.

Reply:
I'm not maintaining this library, so up to you, but I'm with @davewichers  here, I prefer this style. The reason is if this were an really lengthy method, you can tell what is returned by just looking at the return statement and don't need to back up 50 or so lines to look at the declaration. For short methods like the one illustrated, it isn't really necessary but it's not wrong either. Just my $.02.

Reply:
I guess it's just style. I know little to nothing about impact on Java versions so I can only stick with the visual argument.
What I usually do is reducing redundant stuff when writting. Nowadays I don't program much and when I do it's on .NET. I even tend to use `var` when the type is clearly redundant like:
```csharp
var strings = new List<string>;
```
Instead of:
```csharp
List<string> strings = new List<string>;
```
But this is a different language, so I can't tell. From what Kevin commented it seems both ways are "the same". You decide @davewichers.

Reply:
@davewichers So, do you intend to keep the definition of generics?😀

Reply:
> @davewichers So, do you intend to keep the definition of generics?😀

Yes - I would prefer to leave those in. Can you remove those changes from the pull request and then we'll review what's left., which I think @spassarop already has.

Reply:
I did :)

Reply:
done.